### PR TITLE
feat(atomic): added analytics & highlighting to smart snippet suggestions

### PR DESCRIPTION
**Pull request by btaillon**
https://coveord.atlassian.net/browse/KIT-1636

# Smart Snippet footer DOM
image

# Smart Snippet Suggestions footer DOM
image

For some reason, VoiceOver for Mac OS announced the source of the smart snippet as a footer, but announced the source of the smart snippet suggestions as content information. Both are valid IMO.

Thanks to @mbenkirane-coveo for the highlighting!

### CODE REVIEW

File: packages/atomic/src/components/atomic-smart-snippet/atomic-smart-snippet-source.tsx [hunk 1]
```diff
@@ -63,7 +64,7 @@ export class AtomicSmartSnippetSource implements InitializableComponent {
           title={this.source.title}
           href={this.source.clickUri}
           target="_self"
-          className="block mb-6"
```
Comment (btaillon) [inline packages/atomic/src/components/atomic-smart-snippet/atomic-smart-snippet-source.tsx:66]: Only the source in smart snippet needs `mb-6`.

File: packages/atomic/src/components/atomic-smart-snippet/atomic-smart-snippet-source.tsx [hunk 2]
```diff
@@ -46,7 +47,7 @@ export class AtomicSmartSnippetSource implements InitializableComponent {
 
   render() {
     return (
-      <section aria-label={this.bindings.i18n.t('smart-snippet-source')}>
+      <Host>
```
Comment (liviarett) [inline packages/atomic/src/components/atomic-smart-snippet/atomic-smart-snippet-source.tsx:50]: Do you need the `<Host>` if it has only one child?

Comment (btaillon) [reply packages/atomic/src/components/atomic-smart-snippet/atomic-smart-snippet-source.tsx:50]: You don't, but it's not the only child.

File: packages/atomic/src/components/atomic-smart-snippet/atomic-smart-snippet-suggestions.tsx [hunk 3]
```diff
@@ -170,36 +168,31 @@ export class AtomicSmartSnippetSuggestions implements InitializableComponent {
...
-          href={source.clickUri}
-          target="_self"
-          part="source-title"
-          onSelect={() => interactiveResult.select()}
-          onBeginDelayedSelect={() => interactiveResult.beginDelayedSelect()}
-          onCancelPendingSelect={() => interactiveResult.cancelPendingSelect()}
-        >
-          {source.title}
-        </LinkWithResultAnalytics>
+        {relatedQuestion.source && (
```
[the remark this answers is not in the document]
Comment (btaillon) [reply packages/atomic/src/components/atomic-smart-snippet/atomic-smart-snippet-suggestions.tsx:176]: Woops, that's right.

File: packages/atomic/src/components/atomic-smart-snippet/atomic-smart-snippet-source.tsx (hunk 2 above)
Comment (liviarett) [reply packages/atomic/src/components/atomic-smart-snippet/atomic-smart-snippet-source.tsx:50]: where is the second one??

Comment (btaillon) [reply packages/atomic/src/components/atomic-smart-snippet/atomic-smart-snippet-source.tsx:50]: hidden by this button in GitHub:
image

Comment (liviarett) [reply packages/atomic/src/components/atomic-smart-snippet/atomic-smart-snippet-source.tsx:50]: aaah haha I promised I thought I had opened all of these buttons, but missed this one!! sorry

Comment (btaillon) [reply packages/atomic/src/components/atomic-smart-snippet/atomic-smart-snippet-source.tsx:50]: I missed it too for a moment and looked like 5 times, wondering why there was only 1 element. Then I looked at the source code in Visual Studio Code and found it.

**Pull request merged**